src/cores/account/Account.sol

```diff
@@ -0,0 +1,216 @@
+            magicValue = ret;
+        } else {
+            // support non-modular signatures by recovering signer address and reverting malleable or invalid signatures
+            (address signer, ECDSA.RecoverError err) = ECDSA.tryRecover(hash, signature);
+            // return if signature is malformed
+            if (err != ECDSA.RecoverError.NoError) return bytes4(0);
+
+            // authenticate signer
+            bool validSigner = _verifySigner(signer);
+            magicValue = validSigner ? this.isValidSignature.selector : bytes4(hex'ffffffff');
+        }
+    }
+
+    /*===============
+        INTERNALS
+    ===============*/
+
+    /// @dev Function to authenticate a recovered signer address. Identical to `Validator::_verifySigner()`, except
+    /// called only on signatures that were not constructed using the modular verification flag
+    function _verifySigner(address _signer) internal view virtual returns (bool);
+
+    /// @dev View function to limit callers to only the EntryPoint contract of this chain
+    function _checkSenderIsEntryPoint() internal view {
+        if (msg.sender != entryPoint) revert NotEntryPoint(msg.sender);
+    }
+
+    /// @dev Since nonce management and collision checks are handled entirely by the EntryPoint,
+    /// this function is left empty for contracts inheriting from this one to use EntryPoint's defaults
+    /// If sequential `UserOperation` nonce ordering is desired, override this, eg: `require(nonce < type(uint64).max)`
+    function _checkNonce() internal view virtual {}
+
+    /// @dev Function to pre-fund the EntryPoint contract with delta of native currency funds required for a UserOperation
+    /// By default, this function only sends enough funds to complete the current context's UserOperation
+    /// Override if sending custom amounts > `_missingAccountFunds` (or < if reverts are preferrable)
+    function _preFund(uint256 _missingAccountFunds) internal virtual {
+        (bool r, ) = payable(msg.sender).call{ value: _missingAccountFunds }('');
+        require(r);
+    }
+
+    /*===============
+        OVERRIDES
+    ===============*/
+
+    /// @dev Function to withdraw funds using the EntryPoint's `withdrawTo()` function
+    /// @param recipient The address to receive from the EntryPoint balance
+    /// @param amount The amount of funds to withdraw from the EntryPoint
+    function withdrawFromEntryPoint(address payable recipient, uint256 amount) public virtual override onlyOwner {
+        IEntryPoint(entryPoint).withdrawTo(recipient, amount);
+    }
+
+    /// @notice This function must be overridden by contracts inheriting `Account` to delineate 
+    /// the type of Account: `Bot`, `Member`, or `Group`
+    /// @dev Owner stored explicitly using OwnableStorage's ERC7201 namespace
+    function owner() public view virtual override(Access, OwnableInternal) returns (address) {
+        return OwnableInternal.owner();
+    }
```

Comment:
Hmm I was thinking this would be implemented at the level of the non-abstract account implementations like you outlined. By implementing this here, we risk forgetting to override it properly in the 6551 account implementations and add additional bytecode to their deploys without benefit. Was there a reason you wanted this here or can we move it to just `BotAccount`?

Reply:
Moved back to BotAccount- the only reason I moved it up to Account in the inheritance tree was to prevent passing the unimplemented `withdrawFromEntrypoint()` through two levels of inheritance since it should only be available to the owner. 

Doesn't affect anything to have it uninherited through a couple inheritance tiers though, so I fixed by moving it back in cc6de26!